docs/appengine/firebase/tictactoe/main.go

```diff
@@ -0,0 +1,299 @@
+			handleError(w, r, "Could not start game", err)
+			return
+		}
+	} else {
+		// Existing game, join it.
+		if err := datastore.RunInTransaction(ctx, func(ctx context.Context) error {
+			g, err := gameFromRequest(r)
+			if err != nil {
+				return err
+			}
+			game = g
+			if game.UserO == "" {
+				game.UserO = u.ID
+				_, err := datastore.Put(ctx, game.K, game)
+				return err
+			}
+			return nil
+		}, nil); err != nil {
+			handleError(w, r, "Could not join game", err)
+			return
+		}
+	}
+
+	gameKey := game.K.Encode()
+	channelID := u.ID + gameKey
+
+	tok, err := createToken(ctx, channelID)
+	if err != nil {
+		handleError(w, r, "Could not create auth token", err)
+		return
+	}
+
+	d := struct {
+		GameKey        string
+		GameLink       string
+		Me             string
+		Token          string
+		ChannelID      string
+		FirebaseConfig template.HTML
+		InitialMessage *Game
+	}{
+		FirebaseConfig: template.HTML(config),
+		Token:          tok,
+		GameKey:        gameKey,
+		GameLink:       r.URL.Scheme + "://" + r.URL.Host + "/?g=" + gameKey,
+		Me:             u.ID,
+		ChannelID:      channelID,
+		InitialMessage: game,
+	}
+	if err := tmpl.Execute(w, d); err != nil {
+		handleError(w, r, "Could not execute template", err)
+		return
+	}
+}
+
+func createToken(ctx context.Context, channelID string) (string, error) {
+	iss, err := appengine.ServiceAccount(ctx)
+	if err != nil {
+		return "", err
+	}
+	iat := time.Now().Unix()
+	jwt := map[string]interface{}{
+		"iss": iss,
+		"sub": iss,
+		"aud": "https://identitytoolkit.googleapis.com/google.identity.identitytoolkit.v1.IdentityToolkit",
+		"iat": iat,
+		"exp": iat + 3600, // 1 hour
+		"uid": channelID,
+	}
+	body, err := json.Marshal(jwt)
+	if err != nil {
+		return "", err
+	}
+	header := base64.StdEncoding.EncodeToString([]byte(`{"typ":"JWT","alg":"RS256"}`))
+	payload := append([]byte(header), byte('.'))
+	payload = append(payload, []byte(base64.StdEncoding.EncodeToString(body))...)
+	_, sig, err := appengine.SignBytes(ctx, payload)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s.%s", payload, base64.StdEncoding.EncodeToString(sig)), nil
+}
+
+func gameFromRequest(r *http.Request) (*Game, error) {
+	ctx := appengine.NewContext(r)
+
+	k, err := datastore.DecodeKey(r.FormValue("g"))
+	if err != nil {
+		return nil, fmt.Errorf("Invalid game ID: %v", err)
+	}
+	var g Game
+	if err := datastore.Get(ctx, k, &g); err != nil {
+		return nil, err
+	}
+	return &g, nil
+}
+
+func firebase(ctx context.Context) (*firego.Firebase, error) {
+	hc, err := google.DefaultClient(ctx,
+		"https://www.googleapis.com/auth/firebase.database",
+		"https://www.googleapis.com/auth/userinfo.email",
+	)
+	if err != nil {
+		return nil, err
+	}
+	base := os.Getenv("FIREBASE_BASE")
+	if base == "" {
+		// Check the environment variable for the base firebase URL.
+		//
+		// The config should look like:
+		//
+		// env_variables:
+		//    FIREBASE_BASE: https://app-id.firebase.io.com
```

Comment:
Should be `firebaseio.com`, not  `firebase.io.com`.

Also - you didn't want to parse this out of the `firebase_config.html`? :-)